spantest/emulator.go

```diff
@@ -82,11 +82,9 @@ func NewEmulatorFixture(t testing.TB) Fixture {
 	}
 	t.Log("emulator host:", emulatorHost)
 	awaitReachable(t, emulatorHost, 100*time.Millisecond, 10*time.Second)
-	conn, err := grpc.DialContext(
-		ctx,
+	conn, err := grpc.NewClient(
 		emulatorHost,
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithBlock(),
```

Comment:
@alethenorio @thall  I'm not 100% sure about this change. What do you think?
Why do we use the `grpc.WithBlock()` for example?

Reply:
`grpc.WithBlock()` is deprecated.

Had the [same question](https://github.com/einride/cloudrunner-go/pull/659#discussion_r1609798446) myself recently 😂

I think one of the reasons we introduced it was to have any code bugs happen at deployment rather than at runtime because Cloud Run tries to deploy when you create a new revision and if that fails it won't switch to this new revision and without this you'd risk the deployment succeeding and only noticing when a request would come later but I am not even sure that was the case so I don't think it is an issue

Reply:
Okay but shall we then give this change a try?

Reply:
Yes. We did the same in cloudrunner-go